src/cucumber.rs

```diff
@@ -153,7 +154,21 @@ impl<W: World> Cucumber<W> {
         let mut stream = runner.run();
 
         while let Some(event) = stream.next().await {
+            let ret = {
```

Comment:
this is a bit ugly but the only way to keep the result around without cloning every `event`. Cleaner (here) would be to change the `handle_event` to take a reference instead of the value, which would also allow for using more than one handler at a time...

Reply:
Change `handle_event` to take a reference. 👍